src/test/java/bank/PaymentServiceTest.java

```diff
@@ -38,16 +40,90 @@ public void shouldTransferMoneyFromOneAccountToAnother(int accountOneBalance, in
+        accountOne.setBalance(instrumentOne);
+
 
         Account accountTwo = new Account();
+        Instrument instrumentTwo = new Instrument();
         accountTwo.setId(2);
-        accountTwo.setBalance(accountTwoBalance);
+        instrumentTwo.setAmount(accountTwoBalance);
+        accountTwo.setBalance(instrumentTwo);
+
+        Instrument instrumentToTransfer = new Instrument();
+
+        instrumentToTransfer.setAmount(moneyToTransfer);
 
-        testedObject.transferMoney(accountOne,accountTwo,moneyToTransfer);
+        testedObject.transferMoney(accountOne,accountTwo,instrumentToTransfer);
 
-        assertThat(accountOne.getBalance()).isEqualTo(expectedBalanceOfAccountOneAfterTransfer);
-        assertThat(accountTwo.getBalance()).isEqualTo(expectedBalanceOfAccountTwoAfterTransfer);
+        assertThat(accountOne.getBalance().getAmount()).isEqualTo(expectedBalanceOfAccountOneAfterTransfer);
+        assertThat(accountTwo.getBalance().getAmount()).isEqualTo(expectedBalanceOfAccountTwoAfterTransfer);
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowExceptionWhenBalanceTooLow(){
+        Account accountOne = new Account();
+        Instrument instrumentOne = new Instrument();
+        instrumentOne.setAmount(499);
+        instrumentOne.setCurrency(Currency.PLN);
+        accountOne.setId(1);
+        accountOne.setBalance(instrumentOne);
+
+        Account accountTwo = new Account();
+        Instrument instrumentTwo = new Instrument();
+        instrumentTwo.setAmount(200);
+        instrumentTwo.setCurrency(Currency.PLN);
+        accountTwo.setId(2);
+        accountTwo.setBalance(instrumentTwo);
+
+        Instrument instrumentToTransfer = new Instrument();
+        instrumentToTransfer.setAmount(1000);
+        instrumentToTransfer.setCurrency(Currency.PLN);
+
+        testedObject.transferMoney(accountOne,accountTwo,instrumentToTransfer);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldThrowExceptionWhenDifferentCurrenciesTransferred(){
+
+        Account accountOne = new Account();
+        Instrument instrumentOne = new Instrument();
+        instrumentOne.setAmount(1000);
+        instrumentOne.setCurrency(Currency.PLN);
+        accountOne.setId(1);
+        accountOne.setBalance(instrumentOne);
+
+        Account accountTwo = new Account();
+        Instrument instrumentTwo = new Instrument();
+        instrumentTwo.setAmount(200);
+        instrumentTwo.setCurrency(Currency.EUR);
+        accountTwo.setId(2);
+        accountTwo.setBalance(instrumentTwo);
+
+        Instrument instrumentToTransfer = new Instrument();
+        instrumentToTransfer.setAmount(100);
+        instrumentToTransfer.setCurrency(Currency.PLN);
+
+        testedObject.transferMoney(accountOne,accountTwo,instrumentToTransfer);
+    }
+
+//    //mozliwosc sprawdzenia komunikatu rzucanego przez wyjatek (+adnotacja Rule u gory klasy)
+//    @Test
```

Comment:
I assume, that you are storing this commented code as a kind of notes ;) In another cases it shouldn't exist longer in code.